src/AutoMerge/types.jl

```diff
@@ -100,18 +100,30 @@ struct GitHubAutoMergeData
 
     # Environment variables to pass to the subprocess that does `Pkg.add("Foo")` and `import Foo`
     environment_variables_to_pass::Vector{String}
+
+    # Parameters for the linecounts_meet_thresholds guideline:
+    src_min_lines::Integer
+    readme_min_lines::Union{Integer, AbstractFloat}
+    test_min_lines::Union{Integer, AbstractFloat}
+    doc_min_lines::Union{Integer, AbstractFloat}
 end
 
 # Constructor that requires all fields (except `pkg_code_path`) as named arguments.
 function GitHubAutoMergeData(; kwargs...)
     pkg_code_path = mktempdir(; cleanup=true)
     kwargs = (; pkg_code_path=pkg_code_path, kwargs...)
     fields = fieldnames(GitHubAutoMergeData)
-    always_assert(Set(keys(kwargs)) == Set(fields))
+    mismatch = setdiff(Set(keys(kwargs)), Set(fields))
+    if !isempty(mismatch)
+        @warn "Keyword/field mismatch while constructing a GitHubAutoMergeData: $(join(mismatchmissing, ", "))"
+    end
+    always_assert(isempty(mismatch))
     always_assert(kwargs[:authorization] ∈ (:normal, :jll))
     return GitHubAutoMergeData(getindex.(Ref(kwargs), fields)...)
 end
 
+ALL_GUIDELINES = []
```

Comment:
Do we need this?

Reply:
I found it handy to be able to lust and count all of the guidelines.

I don't see that it's doing any harm.

I'm about to submit a cleanup issue where this will play a significant role.

Reply:
Here's that new issue:
https://github.com/JuliaRegistries/RegistryCI.jl/issues/493